gengokit/template/NAME-service/cmd/NAME/main.gotemplate

```diff
@@ -37,12 +37,30 @@ func main() {
 	os.Exit(submain())
 }
 
+type headerSeries []string
+
+func (h *headerSeries) Set(val string) error {
+	requiredPortions := 2
+	parts := strings.SplitN(val, ":", requiredPortions)
+	if len(parts) != requiredPortions {
+		return errors.New(fmt.Sprintf("value %q cannot be split in two; must contain at least one ':' character", val))
+	}
+	*h = append(*h, parts...)
+	return nil
+}
+
+func (h *headerSeries) String() string {
+	return fmt.Sprintf("%v", []string(*h))
+}
+
 // submain exists to act as the functional main, but will return exit codes to
 // the actual main instead of calling os.Exit directly. This is done to allow
 // the defered functions to be called, since if os.Exit where called directly
 // from this function, none of the defered functions set up by this function
 // would be called.
 func submain() int {
+	var headers headerSeries
+	flag.Var(&headers, "header", "Header(s) to be sent in the transport (follows cURL style)")
```

Comment:
Whereas I will take the shorter one every time. I grew up with DOS. Long command line args? Bloated gilding! :P